src/caterpillar_test.py

```diff
@@ -669,5 +678,127 @@ def test_meta_tags(self):
+      for filename in filenames:
+        relpath = os.path.relpath(
+            os.path.join(dirname, filename), self.output_dir)
+        self.assertIn(relpath, expected_files)
+
+  def test_all_expected_output_files(self):
+    """Tests that all expected output files were output."""
+    expected_files = set()
+    for dirname, _, filenames in os.walk(self.output_dir):
+      for filename in filenames:
+        relpath = os.path.relpath(
+            os.path.join(dirname, filename), self.output_dir)
+        expected_files.add(relpath)
+
+    for dirname, _, filenames in os.walk(TTS_REFERENCE_PATH):
+      for filename in filenames:
+        relpath = os.path.relpath(
+            os.path.join(dirname, filename), TTS_REFERENCE_PATH)
+        self.assertIn(relpath, expected_files)
+
+  def test_all_correct_contents(self):
+    """Tests that the content of all non-static output files is expected."""
+    for dirname, _, filenames in os.walk(TTS_REFERENCE_PATH):
+      for filename in filenames:
+        if filename == caterpillar.SW_SCRIPT_NAME:
+          # Service worker is partly random, so test it elsewhere.
+          continue
+
+        reference_path = os.path.join(dirname, filename)
+        relpath = os.path.relpath(reference_path, TTS_REFERENCE_PATH)
+
+        if not (relpath.startswith(self.boilerplate_dir)
+                or relpath.startswith(self.report_dir)):
+          output_path = os.path.join(self.output_dir, relpath)
+          with open(reference_path) as reference_file:
+            with open(output_path) as output_file:
+              output_data = output_file.read().decode('utf-8')
+              reference_data = reference_file.read().decode('utf-8')
+              self.assertEqual(output_data, reference_data,
+                  'Difference found in file `{}`.\n{}'.format(relpath,
+                      '\n'.join(difflib.unified_diff(
+                          output_data.split('\n'),
+                          reference_data.split('\n'),
+                          fromfile=reference_path,
+                          tofile=output_path,
+                          n=0))))
+
+  def test_generated_service_worker(self):
+    """Tests that the generated service worker is as expected."""
+    output_sw_path = os.path.join(self.output_dir, caterpillar.SW_SCRIPT_NAME)
+    reference_sw_path = os.path.join(
+        TTS_REFERENCE_PATH, caterpillar.SW_SCRIPT_NAME)
+
+    with open(output_sw_path) as output_file:
+      with open(reference_sw_path) as reference_file:
+        output_data = output_file.read().decode('utf-8')
+        reference_data = reference_file.read().decode('utf-8')
+
+        # The cache version is randomly generated in the output service worker,
+        # so reset it to be 0, the same cache version as the reference service
+        # worker.
+        output_data = re.sub(
+            r'CACHE_VERSION = \d+', 'CACHE_VERSION = 0', output_data)
+
+        self.assertEqual(output_data, reference_data,
```

Comment:
One thing you could do here is `assertEqual(output_data.split('\n'), reference_data.split('\n'))`, and unittest will automatically show a diff-like output for the two lists.

Not sure if this looks better or worse than the output you've got here. You could experiment and use your judgement.